### DISCUSSION TITLE
How to calculate (relative) complexity of images programmatically?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [dlemstra/Magick.NET#974] [Help]
I have searched the internet thoroughly but not much related questions. Even on stackoverflow there is none

So what I want to achieve is rather hard thing to do

I want to calculate complexity of images

Let me show you several examples from the game I develop https://www.monstermmorpg.com/MonsterDex

A very complex and highly detailed Monster artwork I have e.g.

https://www.monstermmorpg.com/Vesuverex-Monster-Dex-18

[![enter image description here][1]][1]

And another one which is pretty simplistic

https://www.monstermmorpg.com/Cluckoon-Monster-Dex-152

[![enter image description here][2]][2]

So I want to calculate complexity of images. 

There is an example command given in imagemagick forum like this

`magick dragon.png -alpha off -statistic StandardDeviation 5x5 -fill White +opaque Black -format "%[fx:mean]\n`

I want to calculate flat colors complexity, many lines complexity, good light complexity and if there are any other complexities all of them. With having all kind of image complexities I will sort my images and decide which  one fits best to my aim.

**I don't know which complexities are available with Magick.NET but I take all granted.** 

So why do I need? When I resize them with ImageMagick, I will apply filters based on image's complexity such as `unsharp` option

It makes huge difference in borders' aliasing 

  [1]: https://i.stack.imgur.com/151RN.jpg
  [2]: https://i.stack.imgur.com/OGVRj.png

**Comment by FurkanGozukara** [comment]
@dlemstra 

How can I get statistics of non transparent pixels?
```
         FileInfo imgInfo = new FileInfo(vrFile);
                using (MagickImage image = new MagickImage(imgInfo))
                {
                    var vrComposite = image.Statistics().Composite();
                 image.Entropy works but I need to exclude transparent pixels from image
}

```

**Reply by dlemstra** [reply]
Currently this is not possible. You will need to do the math yourself.

```C#
var stats = image.Statistics();
var total = (stats.GetChannel(PixelChannel.Red) +  stats.GetChannel(PixelChannel.Green) +  stats.GetChannel(PixelChannel.Blue)) / 3.0;
```

But in the next release you will be able to do this:

```C#
 var vrComposite = image.Statistics(Channels.RGB).Composite();
```
